### PR TITLE
Suggest CircleCI best practice: step documentation

### PR DESCRIPTION
**Pull request by marians**
Dear SIG releng, with this PR I propose to make CircleCI steps more self-explanatory.

Here is an example of some nicely documented steps, as suggested in the text:

image

And here is an example of what happens if steps are not documented via the `name` field:

image

This is especially important to me, as these automations have to be understood by many people, but not frequently, so there are many people who don't know these steps by heart and would benefit from better readability and explanation, in case something goes wrong.

### CODE REVIEW

**Review by rossf7 — APPROVED** [review]
Nice, good to have this documented. LGTM

File: github/circleci/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# CircleCI best practices
+
+## Making steps self-explanatory
+
+CircleCI offers a way to make a step in a CI config or org self-explanatory. We SHOULD always use this to make the config more self-explanatory.
```
Comment (oponder) [inline github/circleci/README.md]: Can we make this short and sweet with more direct language, and by removing a redundant sentence?

Like:

```suggestion
Always set the `name` of your steps to something that explains what it does.
```

**Comment by kopiczko** [comment]
We already talked about it in the SIG. We want to do that. The downside is that the configuration would be like 3 times longer but I think it's ok price to pay. We also want to prefix steps with job names. Like `architect/push-to-docker: log in to docker`. I'm thinking about moving contents of this PR to `architect-orb` readme because this is where we should define majority of our configs. WDYT?

**Comment by marians** [comment]
@kopiczko Not sure about moving, as it also applies to CircleCI config outside orbs. Rather no. Anyway, I'll merge and may the SIG go from here.

**Comment by marians** [comment]
Well, won't merge, since I need 2 approvals :)

**Review by kopiczko — APPROVED** [review]

**Comment by marians** [comment]
Wait, we need 3 approvals now? Pretty strict :)

**Review by oponder — APPROVED** [review]
LGTM!

**Pull request merged**